File: app/src/main/java/com/monster/app/niceweather/activities/detail/mvp/DetailModel.java
package com.monster.app.niceweather.activities.detail.mvp;

import android.app.Activity;

import com.monster.app.niceweather.activities.detail.DetailActivity;
import com.monster.app.niceweather.models.ForecastCityModel;
import com.monster.app.niceweather.models.ForecastModel;
import com.monster.app.niceweather.models.ListResultResponse;
import com.monster.app.niceweather.network.OpenweatherService;

import io.reactivex.Observable;

/**
 * Created by monster on 2/26/18.
 */

public class DetailModel {

    private final Activity activity;
    private final OpenweatherService service;
    private final ForecastCityModel cityModel;

    public DetailModel(Activity activity, OpenweatherService service) {
        this.activity = activity;
        this.service = service;
        cityModel = (ForecastCityModel) activity.getIntent().getSerializableExtra(DetailActivity.CITY_FORECAST);
    }

    public Observable<ListResultResponse<ForecastModel>> getForecast(int cityId, int count) {
        return service.getForecast(cityId, count);
    }

    public ForecastCityModel getForecastCity() {
        return cityModel;
    }

}

File: app/src/main/java/com/monster/app/niceweather/activities/detail/mvp/DetailModel.kt
package com.monster.app.niceweather.activities.detail.mvp

import android.app.Activity

import com.monster.app.niceweather.activities.detail.DetailActivity
import com.monster.app.niceweather.models.ForecastCityModel
import com.monster.app.niceweather.models.ForecastModel
import com.monster.app.niceweather.models.ListResultResponse
import com.monster.app.niceweather.network.OpenweatherService

import io.reactivex.Observable
import retrofit2.Response

/**
 * Created by monster on 2/26/18.
 */

class DetailModel(private val activity: Activity, private val service: OpenweatherService) {
    val forecastCity: ForecastCityModel

    init {
        forecastCity = activity.intent.getSerializableExtra(DetailActivity.CITY_FORECAST) as ForecastCityModel
    }

    fun getForecast(cityId: Int, count: Int): Observable<ListResultResponse<ForecastModel>> {
        return service.getForecast(cityId, count)
    }

}
